middleware/verify_privilege_token/main.go

```diff
@@ -0,0 +1,64 @@
+package verify_privilege_token
+
+import (
+	"fmt"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/rs/zerolog"
+	"golang.org/x/exp/slices"
+)
+
+type VerifyPrivilegeTokenInterface interface {
+	HasTokenPrivilege(privilegeID int64) fiber.Handler
+}
+
+type Config struct {
+	Log *zerolog.Logger
+}
+
+type verifyPrivilegeToken struct {
+	cfg Config
+}
+
+func New(cfg Config) VerifyPrivilegeTokenInterface {
+	return &verifyPrivilegeToken{
+		cfg: cfg,
+	}
+}
+
+func (p *verifyPrivilegeToken) HasTokenPrivilege(privilegeID int64) fiber.Handler {
+	return func(c *fiber.Ctx) error {
+		return p.checkPrivilege(c, privilegeID)
+	}
+}
+
+func (p *verifyPrivilegeToken) checkPrivilege(c *fiber.Ctx, privilegeID int64) error {
+	logger := p.cfg.Log
+	claims, err := getVehicleTokenClaims(c)
+	if err != nil {
+		logger.Debug().Str("VehicleTokenID In Request", c.Params("tokenID")).
+			Str("VehicleTokenID in bearer token", claims.VehicleTokenID).
+			Msg(err.Error())
```

Comment:
```suggestion
		logger.Debug().Err(err).Str("pathTokenId", tkID).Msg("Couldn't parse JWT.")
```
After moving line 45 up a bit.

I don't think it makes sense to access `claims.VehicleTokenID` if there's an error, it seems like we always return a zero struct in that case.

I like to keep the `.Msg` part fixed so that it's easy to scan for the error in the log reader.